lib/view/movie_detail.dart

```diff
@@ -0,0 +1,201 @@
+import 'package:flutter/material.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'package:movie_app/constain_values/values.dart';
+import 'package:movie_app/view/colors.dart';
+import 'package:movie_app/view/widgets/app_bar_widget.dart';
+import 'package:movie_app/view/widgets/movie_infor_widget.dart';
+import 'package:movie_app/view/widgets/poster_image_widget.dart';
+import 'package:movie_app/view/widgets/text_widget.dart';
+import 'package:movie_app/view_model/movie_detail_view_model.dart';
+import 'package:provider/provider.dart';
+
+class MovieDetail extends StatefulWidget {
+  const MovieDetail({super.key});
+
+  @override
+  State<MovieDetail> createState() => _MovieDetailState();
+}
+
+class _MovieDetailState extends State<MovieDetail> {
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final int movieId = ModalRoute.of(context)?.settings.arguments as int;
+
+    final height = MediaQuery.of(context).size.height;
+    final width = MediaQuery.of(context).size.width;
+    return ChangeNotifierProvider(
+      create: (_) => MovieDetailViewModel(movieId)..fetchMovieDetail(),
+      child: Consumer<MovieDetailViewModel>(
+        builder: (context, viewModel, child) {
+          if (viewModel.movieItem == null && viewModel.isLoading) {
```

Comment:
builder của Consumer sẽ trigger khi mà state thay đổi, đoạn này mình nên hạn chế để toàn bộ UI vào trong Consumer builder, chỉ nên để những cái nó thay đổi thôi nhé 
![image](https://github.com/user-attachments/assets/cffc3a86-4260-4be0-a0b6-baef44ab74cf)